File: navigation/src/main/java/com/example/navigation/stepsEngine/field/InstallmentsField.java
package com.example.navigation.stepsEngine.field;

import com.example.navigation.stepsEngine.payment.FlowState;

public class InstallmentsField extends Field {

    @Override
    public Integer get(FlowState flowState) {
        return flowState.getInstallments();
    }

    @Override
    public String getId() {
        return null;
    }

    @Override
    public void set(FlowState flowState, Object value) {

    }

}

File: navigation/src/main/java/com/example/navigation/stepsEngine/field/InstallmentsField.kt
package com.example.navigation.stepsEngine.field

import com.example.navigation.stepsEngine.payment.FlowState
import kotlinx.android.parcel.Parcelize

@Parcelize
class InstallmentsField : Field() {
    override operator fun get(flowState: FlowState): Int? {
        return flowState.installments
    }

    override fun getId(): String {
        return FieldName.INSTALLMENTS
    }

    override operator fun set(flowState: FlowState, value: Any?) {}
}